### PR TITLE
Updates for building on macos.

### PR DESCRIPTION
**Pull request by tomlandrum**

### CODE REVIEW

File: src/core/Message.cpp [hunk 1]
```diff
@@ -3669,12 +3669,6 @@ class StrategyGetMintResponse : public OTMessageStrategy
         const auto found = String::Factory(xml->getAttributeValue("found"));
         m.m_bBool = found->Compare("true");
 
-        try {
-            m.enum_ = std::stoi(xml->getAttributeValue("type"));
-        } catch (...) {
-            m.enum_ = 0;
-        }
-
```
Comment (justusranvier) [inline src/core/Message.cpp]: Why was this removed?

File: src/network/zeromq/socket/Socket.cpp [hunk 2]
```diff
@@ -50,6 +50,11 @@ Socket::Socket(
     , running_(Flag::Factory(true))
     , type_(type)
 {
+	if (nullptr == socket_) {
+        LogOutput(OT_METHOD)(__FUNCTION__)(": ")(zmq_strerror(zmq_errno()))
+            .Flush();
+	}
+
```
Comment (justusranvier) [inline src/network/zeromq/socket/Socket.cpp]: It's not safe to use Log class here

File: tests/OTTestEnvironment.cpp [hunk 3]
```diff
@@ -6,10 +6,17 @@
 #include "OTTestEnvironment.hpp"
 
 #include "opentxs/opentxs.hpp"
+#include "core/String.hpp"
```
Comment (justusranvier) [inline tests/OTTestEnvironment.cpp]: should be "opentxs/core/String.hpp"

File: src/core/Message.cpp (hunk 1 above)
Comment (tomlandrum) [reply src/core/Message.cpp]: It was causing an unhandled exception on macos.  Since that attribute isn't set in writeXml, I thought it would be okay to just remove it from processXml.

File: src/network/zeromq/socket/Socket.cpp (hunk 2 above)
Comment (tomlandrum) [reply src/network/zeromq/socket/Socket.cpp]: I knew that.

File: tests/OTTestEnvironment.cpp (hunk 3 above)
Comment (justusranvier) [reply tests/OTTestEnvironment.cpp]: Due to how the include paths are set in cmake, #include "core/String.hpp" will match the private header in src/core rather than the public header in include/opentxs/core

File: src/core/Message.cpp (hunk 1 above)
Comment (tomlandrum) [reply src/core/Message.cpp]: It was a segfault, not an unhandled exception.
